pkg/chunk/schema.go

```diff
@@ -204,6 +208,67 @@ func (s schema) GetChunksForSeries(from, through model.Time, userID string, seri
 	return result, nil
 }
 
+// returns IndexEntrys for deletion. It makes sure that we don't include series entries which are in use by verifying using hasChunksForIntervalFunc
+func (s schema) GetSeriesDeleteEntries(from, through model.Time, userID string, metric labels.Labels, hasChunksForIntervalFunc hasChunksForIntervalFunc) ([]IndexEntry, error) {
+	buckets := s.buckets(from, through, userID)
+	if len(buckets) == 0 {
+		return nil, nil
+	}
+
+	seriesID := string(labelsSeriesID(metric))
+
+	// Only first and last buckets needs to be checked for in-use series ids.
+	// Only partially deleted first/last deleted bucket needs to be checked otherwise
+	// not since whole bucket is anyways considered for deletion.
+	if buckets[0].from != 0 {
+		bucketStartTime := from - model.Time(buckets[0].from)
```

Comment:
from would always be bigger because `bucket.from` is relative i.e for a per-day bucket it would be the number of seconds elapsed since the start of that day. Sounds good?